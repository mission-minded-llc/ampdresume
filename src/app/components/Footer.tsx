"use client";

import { BottomNavigation, Box, Typography } from "@mui/material";

import { MuiLink } from "@/components/MuiLink";
import React from "react";
import { ThemeAppearanceToggle } from "./ThemeAppearanceToggle";
import { getBaseUrl } from "@/util/url";
import { useIsDesktop } from "@/hooks/useIsDesktop";
import { useIsLoggedIn } from "@/hooks/useIsLoggedIn";
import { useIsResumePage } from "@/hooks/useIsResumePage";

export const Footer = () => {
  const baseUrl = getBaseUrl();
  const isDesktop = useIsDesktop();
  const isLoggedIn = useIsLoggedIn();
  const isResumePage = useIsResumePage();

  return (
    <BottomNavigation
      component="footer"
      sx={(theme) => ({
        backgroundColor: theme.palette.background.paper,
        position: isResumePage ? "fixed" : "relative",
        marginTop: "auto",
        height: "auto",
        overflow: "auto",
        zIndex: 2,
        bottom: 0,
        left: 0,
        width: "100%",
        boxShadow: "0 0 35px rgba(0, 0, 0, 0.1)",
      })}
    >
      <Box
        sx={{
          width: "100%",
          display: "flex",
          flexDirection: isDesktop ? "row" : "column",
          padding: "0.5em",
          justifyContent: "space-around",
          alignItems: "center",
          textAlign: "center",
        }}
      >
        <Box
          sx={{
            display: "flex",
            flexDirection: "row",
            justifyContent: "space-around",
            alignItems: "center",
            textAlign: "center",
            width: "100%",
            pl: 2,
            gap: "0.35em",
          }}
        >
          {isDesktop && !isLoggedIn ? "Create your free interactive resume at " : null}
          <MuiLink href={baseUrl}>openresume.org</MuiLink>
          {isResumePage ? <ThemeAppearanceToggle /> : null}
        </Box>
        <Box>
          <Typography variant="caption" sx={{ textAlign: "center" }}>
            <MuiLink href="/about/privacy-policy">Privacy Policy</MuiLink> |{" "}
            <MuiLink href="/about/terms-of-service">Terms of Service</MuiLink>
            {isDesktop ? (
              <> | &copy; {new Date().getFullYear()} OpenResume. All rights reserved.</>
            ) : null}
          </Typography>
        </Box>
      </Box>
    </BottomNavigation>
  );
};
