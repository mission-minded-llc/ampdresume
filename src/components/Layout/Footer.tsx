"use client";

import React from "react";

import { BottomNavigation, Box } from "@mui/material";
import { getBaseUrl } from "@/util/url";
import { usePathname } from "next/navigation";
import { ThemeAppearanceToggle } from "./ThemeAppearanceToggle";
import { MuiLink } from "../MuiLink";
import { useIsDesktop } from "@/hooks/useIsDesktop";
import { useIsLoggedIn } from "@/hooks/useIsLoggedIn";

export const Footer = () => {
  const baseUrl = getBaseUrl();
  const pathname = usePathname();
  const isDesktop = useIsDesktop();
  const isLoggedIn = useIsLoggedIn();

  const isResumePage = pathname.startsWith("/r/");
  const isHomePage = pathname === "/";
  const isLoginPage = pathname === "/login";

  if (isHomePage || isLoginPage) return null;

  return (
    <BottomNavigation
      component="footer"
      sx={(theme) => ({
        backgroundColor: theme.palette.background.paper,
        position: isResumePage ? "fixed" : "relative",
        zIndex: 2,
        bottom: 0,
        left: 0,
        width: isResumePage ? "100vw" : "auto",
        boxShadow: "0 0 35px rgba(0, 0, 0, 0.1)",
      })}
    >
      <Box
        sx={{
          width: "100%",
          display: "flex",
          justifyContent: "space-around",
          alignItems: "center",
        }}
      >
        <Box
          sx={{
            display: "flex",
            alignItems: "center",
            gap: "0.35em",
          }}
        >
          {isDesktop && !isLoggedIn ? `Create your free interactive resume at ` : null}
          <MuiLink href={baseUrl}>openresume.org</MuiLink>
        </Box>
        {isResumePage ? <ThemeAppearanceToggle /> : null}
      </Box>
    </BottomNavigation>
  );
};
