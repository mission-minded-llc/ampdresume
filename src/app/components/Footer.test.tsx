import "@testing-library/jest-dom";

import { render } from "@testing-library/react";
import { usePathname } from "next/navigation";
import React from "react";

import { Footer } from "./Footer";

jest.mock("next/navigation");

describe("Footer component", () => {
  it("matches snapshot for resume page", () => {
    (usePathname as jest.Mock).mockReturnValue("/r/john-doe"); // Resume page.

    const { container } = render(<Footer />);
    expect(container).toMatchSnapshot();

    // Expect the footer element to be in fixed position.
    const footer = container.querySelector("footer");
    expect(footer).toHaveStyle("position: fixed");
  });

  it("matches snapshot for regular page", () => {
    (usePathname as jest.Mock).mockReturnValue("/about"); // Non-resume page.

    const { container } = render(<Footer />);
    expect(container).toMatchSnapshot();

    // Expect the footer element to be in fixed position.
    const footer = container.querySelector("footer");
    expect(footer).toHaveStyle("position: relative");
  });
});
