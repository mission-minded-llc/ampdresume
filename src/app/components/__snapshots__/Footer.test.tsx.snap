// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer component matches snapshot for regular page 1`] = `
<div>
  <footer
    class="MuiBox-root css-1xsyhud"
  >
    <div
      class="MuiBox-root css-1a8n5hv"
    >
      <a
        class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1orsvah-MuiTypography-root-MuiLink-root"
        href="http://localhost:3000"
        target="_self"
      >
        ampdresume.com
      </a>
      <div
        class="MuiBox-root css-0"
      >
        <span
          class="MuiTypography-root MuiTypography-caption css-1fi1ecl-MuiTypography-root"
        >
          <a
            class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1orsvah-MuiTypography-root-MuiLink-root"
            href="/about/privacy-policy"
            target="_self"
          >
            Privacy Policy
          </a>
           |
           
          <a
            class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1orsvah-MuiTypography-root-MuiLink-root"
            href="/about/terms-of-service"
            target="_self"
          >
            Terms of Service
          </a>
        </span>
      </div>
    </div>
  </footer>
</div>
`;

exports[`Footer component matches snapshot for resume page 1`] = `
<div>
  <footer
    class="MuiBox-root css-1417pob"
  >
    <div
      class="MuiBox-root css-1a8n5hv"
    >
      <a
        class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1orsvah-MuiTypography-root-MuiLink-root"
        href="http://localhost:3000"
        target="_self"
      >
        ampdresume.com
      </a>
      <div
        class="MuiBox-root css-0"
      >
        <span
          class="MuiTypography-root MuiTypography-caption css-1fi1ecl-MuiTypography-root"
        >
          <a
            class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1orsvah-MuiTypography-root-MuiLink-root"
            href="/about/privacy-policy"
            target="_self"
          >
            Privacy Policy
          </a>
           |
           
          <a
            class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1orsvah-MuiTypography-root-MuiLink-root"
            href="/about/terms-of-service"
            target="_self"
          >
            Terms of Service
          </a>
        </span>
      </div>
    </div>
  </footer>
</div>
`;
